src/test/java/cloud/fogbow/ras/core/plugins/interoperability/opennebula/attachment/v5_4/OpenNebulaAttachmentPluginTest.java

```diff
@@ -150,277 +139,349 @@ public void testHasFailedUnsuccessful() {
 		}
 	}
 	
-	// test case: When invoking the requestInstance method, with the valid client
+	// test case: When invoking the requestInstance method, with a valid client
 	// and template, a virtual machine will be instantiated to attach a volume image
-	// disk, returning the attached disk ID in conjunction with the other IDs of
-	// that instance.
+	// disk and return the attached disk ID
 	@Test
-	public void testRequestInstanceSuccessful() throws FogbowException {
+	public void testRequestInstance() throws FogbowException {
 		// set up
-		VirtualMachine virtualMachine = Mockito.mock(VirtualMachine.class);
-		PowerMockito.mockStatic(OpenNebulaClientUtil.class);
-		BDDMockito.given(OpenNebulaClientUtil.getVirtualMachine(Mockito.any(Client.class), Mockito.anyString()))
-				.willReturn(virtualMachine);
-
-		String template = generateAttachmentTemplate();
-		OneResponse response = Mockito.mock(OneResponse.class);
-		Mockito.when(virtualMachine.diskAttach(Mockito.contains(template))).thenReturn(response);
-		Mockito.when(response.isError()).thenReturn(false);
-
-		Mockito.when(virtualMachine.info()).thenReturn(response);
-		Mockito.when(response.getMessage()).thenReturn(VIRTUAL_MACHINE_CONTENT);
-
-		CloudUser cloudUser = createCloudUser();
+		Mockito.when(this.virtualMachine.diskAttach(this.template)).thenReturn(this.response);
+		Mockito.when(this.response.isError()).thenReturn(false);
 
 		// exercise
-		this.plugin.requestInstance(attachmentOrder, cloudUser);
+		this.plugin.requestInstance(this.attachmentOrder, this.cloudUser);
```

Comment:
this test should mock doRequestInstance() and getDiskIdFromContentOf() and below, create separate tests for getDiskIdFromContentOf().

Reply:
Same for deleteInstance and getInstance related methods.

Reply:
done.